include/sched.h

```diff
@@ -0,0 +1,70 @@
+#ifndef SCHED_H
+#define SCHED_H
+
+#include "ultra64.h"
+#include "irqmgr.h"
+
+#define OS_SC_NEEDS_RDP     0x0001  // Task uses the RDP
+#define OS_SC_NEEDS_RSP     0x0002  // Task uses the RSP
+#define OS_SC_DRAM_DLIST    0x0004  // RDP commands generated in the RSP pass through DRAM
+#define OS_SC_PARALLEL_TASK 0x0010  // Unimplemented
+#define OS_SC_LAST_TASK     0x0020  // Unimplemented
+#define OS_SC_SWAPBUFFER    0x0040  // Swap framebuffer when done
+
+#define OS_SC_DP        0x0001  // Task is using the RDP
+#define OS_SC_SP        0x0002  // Task is using the RSP
+#define OS_SC_YIELD     0x0010  // Task has been asked to yield
+#define OS_SC_YIELDED   0x0020  // Task has yielded
+
+#define OS_SC_RCP_MASK  (OS_SC_NEEDS_RDP | OS_SC_NEEDS_RSP)
+#define OS_SC_TYPE_MASK (OS_SC_NEEDS_RDP | OS_SC_NEEDS_RSP | OS_SC_DRAM_DLIST)
+
+typedef struct {
+    /* 0x00 */ u16* framebuffer;    // current framebuffer
+    /* 0x04 */ u16* swapBuffer;     // framebuffer to swap to
+    /* 0x08 */ OSViMode* viMode;
+    /* 0x0C */ u32 viFeatures;
+    /* 0x10 */ u8 unk_10;           // set to 0, never read
+    /* 0x11 */ s8 updateRate;       // how many VIs should elapse before next swap
+    /* 0x12 */ s8 updateTimer;      // counts down from updateRate to 0, swaps the framebuffer at 0
```

Comment:
because this is OoT and we expect timers to be in frames instead of retrace counts (this is in retrace counts right?),
maybe this could be more explicit like `retracesUntilFbSwap` ? I can't find a great name...

(later) `retracesCountDown`? `viTimer`, `retraceTimer`?

Reply:
Not sure, practically any timer that is outside gamestate code will be counted in retraces, so it might be impractical to be overly verbose with them all. I think we may need to consider this with the bigger picture taken into account.